web/docker/entrypoint.sh

```diff
@@ -1,29 +1,41 @@
 #!/bin/bash
 
-USER_ID=$($USER)
+USER_ID="$(id -u)"
 USER_GROUP="$(id -g)"
 WORKSPACE_DIR="/workspace"
 PGPASS_FILE=$WORKSPACE_DIR/.pgpass
 
-if [ "$(id -u)" = '0' ]; then
-  # Create workspace directory and change the permission to 'codechecker' user
-  # if this directory doesn't exists.
-  if [ ! -d $WORKSPACE_DIR ]; then
+if [ "$USER_ID" = '0' ]; then
+  echo "Container started with 'root' user."
+
+  if [ -d $WORKSPACE_DIR ]; then
+    echo "Workspace directory '${WORKSPACE_DIR}' already exists."
+
+    workspace_dir_owner=$(stat ${WORKSPACE_DIR} -c %u)
+    if [ "${workspace_dir_owner}" != '0' ]; then
+      echo "Executing script with workspace directory owner (UID): '${workspace_dir_owner}'..."
+      exec gosu $workspace_dir_owner "$0" "$@"
+    fi
+  else
+    echo "Creating workspace directory: '${WORKSPACE_DIR}'."
+    echo "WARNING: This directory exists ONLY within the containter!"
+
     mkdir -p $WORKSPACE_DIR
     chown -R codechecker:codechecker $WORKSPACE_DIR
-  fi
 
-  # Execute this script again with codechecker user.
-  exec gosu codechecker "$0" "$@"
+    echo "Executing script with internal 'codechecker' user (UID: $USER_ID)..."
```

Comment:
```
$ docker run codechecker-web:csordasmarton/fix_docker_entry_point --help
Container started with 'root' user.
Creating workspace directory: '/workspace'.
WARNING: This directory exists ONLY within the containter!
Executing script with internal 'codechecker' user (UID: 0)...
Executing command: '--help'.
exec: exec [-cl] [-a name] [command [arguments ...]] [redirection ...]
```

I think here you shouldn't say `$USER_ID`, but `$(id -u codechecker)`? Because this line is executed **before** the `exec` call.

Reply:
Good catch, fixed it.